internal/cephfs/volume.go

```diff
@@ -94,36 +83,45 @@ func getVolumeRootPathCeph(ctx context.Context, volOptions *volumeOptions, cr *u
 	return strings.TrimSuffix(stdout, "\n"), nil
 }
 
-func getSubVolumeInfo(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID) (Subvolume, error) {
-	info := Subvolume{}
-	err := execCommandJSON(
-		ctx,
-		&info,
-		"ceph",
-		"fs",
-		"subvolume",
-		"info",
-		volOptions.FsName,
-		string(volID),
-		"--group_name",
-		volOptions.SubvolumeGroup,
-		"-m", volOptions.Monitors,
-		"-c", util.CephConfigPath,
-		"-n", cephEntityClientPrefix+cr.ID,
-		"--keyfile="+cr.KeyFile)
+func (vo *volumeOptions) getSubVolumeInfo(ctx context.Context, volID volumeID) (*Subvolume, error) {
+	fsa, err := vo.conn.GetFSAdmin()
+	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can not fetch metadata pool for %s:", vo.FsName, err)
+		return nil, err
+	}
+
+	info, err := fsa.SubVolumeInfo(vo.FsName, vo.SubvolumeGroup, string(volID))
 	if err != nil {
 		util.ErrorLog(ctx, "failed to get subvolume info for the vol %s: %s", string(volID), err)
 		if strings.HasPrefix(err.Error(), volumeNotFound) {
-			return info, ErrVolumeNotFound
+			return nil, ErrVolumeNotFound
 		}
 		// Incase the error is other than invalid command return error to the caller.
 		if !strings.Contains(err.Error(), invalidCommand) {
-			return info, ErrInvalidCommand
+			return nil, ErrInvalidCommand
 		}
 
-		return info, err
+		return nil, err
 	}
-	return info, nil
+
+	var bq int64
+	bc, ok := info.BytesQuota.(fsAdmin.ByteCount)
+	if !ok {
```

Comment:
we are not supporting infinite, isn't it? how about returning an error message if this fails.

Reply:
Sure, that makes things a little simpler. It is not clear from the CSI spec if 'infinite' (or unset capacity) is a common option. Will drop it with the next update.